ee/clickhouse/queries/funnels/funnel_correlation.py

```diff
@@ -415,16 +431,41 @@ def _get_events_join_query(self) -> str:
                 AND event.event NOT IN funnel_step_names
         """
 
+    def _get_aggregation_join_query(self):
+        if self._filter.aggregation_group_type_index is None:
+            person_query, person_query_params = ClickhousePersonQuery(
+                self._filter, self._team.pk, ColumnOptimizer(self._filter, self._team.pk)
+            ).get_query()
+
+            return (
+                f"""
+                JOIN ({person_query}) person
+                    ON person.id = funnel_people.person_id
+            """,
+                person_query_params,
+            )
+        else:
+            return GroupsJoinQuery(
+                self._filter, self._team.pk, ColumnOptimizer(self._filter, self._team.pk)
```

Comment:
Nit: It's worth pulling this out into the constructor!

Reply:
and same for `ClickhousePersonQuery` ?

Reply:
Ah, you mean the `GroupsJoinQuery` construction?

Reply:
I meant ColumnOptimizer construction

Reply:
gotcha, yes, that makes more sense